### PR TITLE
[MAJOR] Implementing transactions management feature.

### LINKED ISSUE #197: Transaction Project Feature implementation
Epic/User story #160

### PR DESCRIPTION
**Pull request by Mnx458**
#160 Transaction feature

closes #197

### CODE REVIEW

**Review by nasaku898 — APPROVED** [review]

**Review by Manraj45 — COMMENTED** [review]
I have a few comments.

File: backend/src/main/routes/TransactionRoute.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  ) {
+    super(app, 'TransactionRoute');
+  }
+
+  configureRoutes(): express.Application {
+    this.getApp()
+      .route(`/transactions/expenses`)
+      .all(
+        checkJwt
+        // checkRole(new Set([Roles.SUPERVISOR, Roles.BUSINESS]))
```
Comment (Manraj45) [inline backend/src/main/routes/TransactionRoute.ts:26]: All the checkroles in this files are also commented out?

File: backend/src/main/routes/ProjectRoute.ts [hunk 2]
```diff
@@ -44,11 +45,32 @@ export default class ProjectRoute extends CommonRoutesConfig {
         }
       });
 
+    this.getApp()
+      .route(`/project/booked`)
+      .get(
+        checkJwt,
+        // checkRole(new Set([Roles.SUPERVISOR, Roles.BUSINESS])),
```
[the remark this answers is not in the document]
Comment (Mnx458) [reply backend/src/main/routes/ProjectRoute.ts:52]: There seems to be an issue with auth0, sometimes our backend hangs with a 429 error but it stays alive. We will rethink how to implement the roles verification later.

**Review by Manraj45 — APPROVED** [review]
Good job! @nasaku898 we should create an issue for the check role.

**Pull request merged**